prober/http.go

```diff
@@ -128,7 +200,7 @@ func ProbeHTTP(ctx context.Context, target string, module config.Module, registr
 		targetHost = targetURL.Host
 	}
 
-	ip, err := chooseProtocol(module.HTTP.PreferredIPProtocol, targetHost, registry, logger)
+	ip, err := chooseProtocolMetrics(module.HTTP.PreferredIPProtocol, targetHost, durationGaugeVec.WithLabelValues("resolve"), probeIPProtocolGauge, logger)
```

Comment:
This removes the existing duration gauge, which needs to stay around. If you need the duration, get the function to pass it back rather than splitting the logic in two and being inconsistent.

Reply:
Yes, the idea was to use the http durationGauge with 'resolve' label for that instead.
I could expose both?
I definitely would want to have the resolution in the time in the overall http duration metric. It's just so much more convienient to consume.